packages/dom/src/dom/replace-tag.js

```diff
@@ -13,6 +18,7 @@ export default function replaceTag( node, tagName ) {
 		newNode.appendChild( node.firstChild );
 	}
 
+	assertIsDefined( node.parentNode, 'node.parentNode' );
```

Comment:
One thing I was contemplating recently when skimming PRs that add types is whether `assertIsDefined` should be executed in the development mode only – effectively it would have all logic behind the `process.env.NODE_ENV`. Would it make sense to optimize it this way?

Reply:
Maybe? Are you suggesting merely updating the body of `assertIsDefined` to the following:

export function assertIsDefined< T >(
	val: T,
	name: string
): asserts val is NonNullable< T > {
	if ( process.env.NODE_ENV === 'development' && ( val === undefined || val === null ) ) {
		throw new Error(
			`Expected '${ name }' to be defined, but received ${ val }`
		);
	}
}

Reply:
Do you prefer to see the error to be thrown on production?

Reply:
Hmmm, I think I'm undecided on whether it's useful in production 🙂 I'll defer to you on that, happy to open a PR to make the change so that dead code elimination works on it.

Reply:
Done! https://github.com/WordPress/gutenberg/pull/30807

Reply:
We can change it later for sure or leave it as is. The question how much of the value it is on production to see a formatted error vs the actual error. In both cases, it's too late to do anything  useful for the user experience 😅

Maybe it doesn't matter.